src/include/serial_inline.h

```diff
@@ -323,7 +323,14 @@ __wt_update_serial(WT_SESSION_IMPL *session, WT_CURSOR_BTREE *cbt, WT_PAGE *page
         page->modify->obsolete_check_txn = WT_TXN_NONE;
     }
 
-    __wt_update_obsolete_check(session, cbt, upd->next, true);
+    /*
+     * Look for obsolete updates if:
+     * 1. The page is configured to evict soon
+     * 2. There are several updates in the cache that need to be removed.
+     */
+    if (WT_READGEN_EVICT_SOON(page->read_gen) ||
```

Comment:
Will we ever have an "EVICT_SOON" page here? I would expect that updating a page causes WT to increase it's eviction generation.

Reply:
Yes, it is possible that a page with EVICT_SOON can come here. We ignore bumping the `page->read_gen` value if the page is set for force eviction. Check the function `__wt_cache_read_gen_bump`.

Reply:
Ah. I hadn't spotted that. Thanks for the clarification, @kommiharibabu